### PR TITLE
add-workflow-wait-for-signals-with-variables-and-refactor

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

File: GetStarted/resources/catalog/wait_for_signals_with_variables.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="urn:proactive:jobdescriptor:3.13" xsi:schemaLocation="urn:proactive:jobdescriptor:3.13 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.13/schedulerjob.xsd"  name="Wait_For_Signal_With_Variables" projectName="2. Advanced Workflows" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <description>
+    <![CDATA[ A template workflow that waits until one signal that is specified in the variable SIGNAL with variables is added to the set of job signals. ]]>
```
Comment (aminelouati) [inline GetStarted/resources/catalog/wait_for_signals_with_variables.xml:6]: one signal that is specified in the variable SIGNAL with variables --> one signal with variables that is specified in the variable SIGNAL

File: GetStarted/resources/catalog/wait_for_signals_with_variables.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  <taskFlow>
+    <task name="wait_for_any_signal_with_variables"
+
+
+
+
+          fork="true">
+      <description>
+        <![CDATA[ A template task that waits until one signal specified in the variable SIGNAL with variables is added to the set of job signals. ]]>
+      </description>
```
Comment (aminelouati) [inline GetStarted/resources/catalog/wait_for_signals_with_variables.xml:22]: same as above

File: GetStarted/resources/catalog/wait_for_signals_with_variables.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      <variables>
+        <variable name="SIGNAL" value="stop" inherited="false"     />
```
Comment (aminelouati) [inline GetStarted/resources/catalog/wait_for_signals_with_variables.xml:24]: can you add a variable description. Please try to do that for all variables. Thanks

File: GetStarted/resources/catalog/wait_for_signals_with_variables.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+
+
+
+          fork="true">
+      <description>
+        <![CDATA[ A template task that waits until one signal with variables that is specified in the variable SIGNAL is added to the set of job signals. ]]>
+      </description>
+      <variables>
+        <variable name="SIGNAL" value="stop" inherited="false" description="Signal that is to be added to the set of job signals."/>
```
Comment (mboussaa) [inline GetStarted/resources/catalog/wait_for_signals_with_variables.xml:24]: is to be added => is added? or that will be added

Comment (oanaBiancaSchiopu) [reply GetStarted/resources/catalog/wait_for_signals_with_variables.xml:24]: I wanted to say "is to be added"  meaning "should be added". I can change it to "will be added"

File: GetStarted/resources/catalog/wait_for_signals_with_variables.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="urn:proactive:jobdescriptor:3.13" xsi:schemaLocation="urn:proactive:jobdescriptor:3.13 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.13/schedulerjob.xsd"  name="Wait_For_Signal_With_Variables" projectName="2. Advanced Workflows" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <description>
+    <![CDATA[ A template workflow that waits until one signal with variables that is specified in the variable SIGNAL is added to the set of job signals. ]]>
```
Comment (nebilbenmabrouk) [inline GetStarted/resources/catalog/wait_for_signals_with_variables.xml:6]: I suggest:
A template workflow that waits until one signal (with input parameters) that is specified in the variable SIGNAL is added to the set of job signals.

File: Signal/resources/catalog/check_and_remove_many_signals.groovy [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+//Read the variable SIGNALS_SET
+Set signalsSet =  variables.get("SIGNALS_SET")
+
+// Check whether one signal among those specified as input is received
+println("Checking whether one signal in the set "+ signalsSet +" is received")
+receivedSignals = signalapi.checkForSignals(signalsSet)
+
+// If a signal is received, remove ready signals and break the loop, else sleep 10 seconds then restart
```
Comment (nebilbenmabrouk) [inline Signal/resources/catalog/check_and_remove_many_signals.groovy:28]: I suggest removing the last part of the sentence, like this:
// If a signal is received, remove ready signals and break the loop

**Review by nebilbenmabrouk — APPROVED** [review]

**Review by mboussaa — APPROVED** [review]

**Review by aminelouati — APPROVED** [review]

**Pull request merged**